pkgs/by-name/ob/obj2voxel/package.nix

```diff
@@ -27,10 +28,28 @@ stdenv.mkDerivation (finalAttrs: {
   );
 
   patches = [
-    # pull missing definitions (https://github.com/Eisenwave/voxel-io/pull/3)
-    ./stdlib-defs.patch
+    # pull missing definitions
+    (
+      fetchpatch {
+        name = "add-std-includes.patch";
+        url = "https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/3.diff";
+        hash = "sha256-2JeFBx0Ds++BplWXuxOo4KOAW6gXCT/kFI74i0M0ROM=";
+        stripLen = 1;
+        extraPrefix = "voxelio/";
+      }
+    )
     # fix compile error on aarch64-darwin
-    ./fix-type-mismatch.patch
+    (
+      fetchpatch {
+        name = "fix-aarch64-darwin-type-mismatch.patch";
+        url = "https://patch-diff.githubusercontent.com/raw/Eisenwave/voxel-io/pull/4.diff";
```

Comment:
Use a commit url like https://github.com/Eisenwave/voxel-io/commit/2f800f6aca8acd0cc18f9625b411c6dc2956f2ea.patch because a push to the PR branch would invalidate the hash.

Reply:
Using a pull-related patch URL was suggested earlier, and I actually think it makes sense. In case the PR would change, it is likely that the changes are also useful for the maintenance of the nix package, thus justifying an update of the hash.

Reply:
We want to be able to build this package in the far future too so we need stable URLs.

Reply:
Being able to build packages in the far future is a formidable goal. I doubt this is the right place to achieve it, though. It's not how Nix package are usually designed. E.g. even with git sources that could be potentially long-term stable, packages tend to reference (potentially unstable) tags instead of cryptographic hashes. I often encounter packages where the URL resource can't be retrieved anymore, or the hash has changed. Which shows that Nix has decided to take a compromise here. Stable URLs are being considered, but may be avoided if there is a reason to.

There is also a reason to reference pull requests instead of commit hashes because pull requests also refer to a specific goal with respect to the upstream source, and possibly the discussion around it. This helps maintainers because later on when changing the package, they can decide whether that specific goal is still relevant.

Nonetheless, the URLs have been changed now to URLs with the objective to remain stable over time. I hope this can relieve your concerns, while still keeping the possibility to look up the related pull request easily.

Reply:
You can and should always reference the relevant PR and describe what it does in a comment right next to the patch URL, again best of both worlds.

Reply:
Added that comment. The stability of commit diff URLs should still not be overestimated, I also saw them disappear in case of force-pushes sometimes.